### PR TITLE
 [release-2.1 < T0769-MG] Add retry transaction docs 

### PR DESCRIPTION
**Pull request by kostasrim**

### CODE REVIEW

File: docs/reference-guide/streams/overview.md [hunk 1]
```diff
@@ -140,3 +140,23 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
 semantics, we tried to minimize the possibility of processing messages multiple
 times. This means committing the message offsets to the Kafka cluster happens
 right after the transaction is committed to the database.
+
+## Configuring stream transactions
+
+A stream fails when a transformation fails to commit a transaction. This is a side effect
+of [isolation levels](/reference-guide/isolation-levels.md) and its behaviour can be
+configured by setting the appropriate memgraph flag:
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/streams/overview.md:148]: It suggests me that a transaction can only fail because of conflict, but that is not true. A transaction can fail because of various reasons. I would rephrase this as:
```suggestion
A stream might fail when a transaction (in which the returned queries are executed) fails to commit because of another conflicting transaction. This is a side effect
of [isolation levels](/reference-guide/isolation-levels.md) and its behaviour can be
configured by setting the appropriate memgraph flag:
```

It is a bit rough, so feel free to rephrase it.

**Review by antaljanosbenjamin — COMMENTED** [review]
Please also update the [configuration page](https://github.com/memgraph/docs/blob/T0769-MG-add-retry-transaction-docs/docs/reference-guide/configuration.md). As we adding more and more streaming related config flags, feel free to create a new section for streaming.

File: docs/reference-guide/streams/overview.md [hunk 2]
```diff
@@ -140,3 +140,23 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
...
+
+```
+--stream-transaction-conflict-retries TIMES_TO_RETRY
+```
+
+By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
+to 0 then memgraph tries a transaction only once. However, for any other value,
+memgraph retries with a total number of: one + `TIMES_TO_RETRY`.
```
Comment (jbajic) [inline docs/reference-guide/streams/overview.md:156]:
```suggestion
By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
to 0 then memgraph tries a transaction only once. However, for any other value,
memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`. 
```

File: docs/reference-guide/streams/overview.md [hunk 3]
```diff
@@ -140,3 +140,23 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
...
+
+```
+--stream-transaction-conflict-retries TIMES_TO_RETRY
+```
+
+By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
+to 0 then memgraph tries a transaction only once. However, for any other value,
+memgraph retries with a total number of: one + `TIMES_TO_RETRY`. 
+
+Morever, the interval of retries is also important and can be configured by
```
Comment (jbajic) [inline docs/reference-guide/streams/overview.md:158]:
```suggestion
Moreover, the interval of retries is also important and can be configured by 
```

File: docs/reference-guide/streams/overview.md [hunk 4]
```diff
@@ -140,3 +140,23 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
...
+
+By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
+to 0 then memgraph tries a transaction only once. However, for any other value,
+memgraph retries with a total number of: one + `TIMES_TO_RETRY`. 
+
+Morever, the interval of retries is also important and can be configured by 
+```
+--stream-transaction-retry_interval INTERVAL_TIME
+```
+The `INTERVAL_TIME` is meassured in `milliseconds` and its default value is `500ms`.
```
Comment (jbajic) [inline docs/reference-guide/streams/overview.md:162]:
```suggestion
The `INTERVAL_TIME` is measured in `milliseconds` and its default value is `500ms`.
```

**Review by g-despot — APPROVED** [review]

File: docs/reference-guide/streams/overview.md [hunk 5]
```diff
@@ -140,3 +140,24 @@ the Kafka cluster. However, even though we cannot guarantee **exactly once**
...
+transaction. This is a side effect of [isolation levels](/reference-guide/isolation-levels.md) 
+and its behaviour can be configured by setting the appropriate memgraph flag:
+
+```
+--stream-transaction-conflict-retries=TIMES_TO_RETRY
+```
+
+By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
+to 0 then memgraph tries a transaction only once. However, for any other value,
+memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`.
```
Comment (antonio2368) [inline docs/reference-guide/streams/overview.md:157]:
```suggestion
By default, memgraph retries a transaction 31 times. If `TIMES_TO_RETRY` is set
to 0 then memgraph tries a transaction only once. However, for any other value,
memgraph retries with a total number of: 1 + `TIMES_TO_RETRY`. 
```
Weird phrasing of it. 31 seems like a strange number.
```
Memgraph will always try to execute the transaction once. 
If it fails because of conflicts with other transactions it will retry the transaction up to `TIMES_TO_RETRY` times. 
By default, that number is set to 30....
```

File: docs/reference-guide/configuration.md [hunk 6]
```diff
@@ -52,6 +52,12 @@ Each configuration setting is in the form: `--setting-name=value`.
 | --storage-wal-file-flush-every-n-tx=100000 | Issue a 'fsync' call after this amount of transactions are written to the WAL file. Set to 1 for fully synchronous operation. | `[uint64]` |
 | --storage-wal-file-size-kib=20480 | Minimum file size of each WAL file. | `[uint64]` |
 
+## Streams
+| Flag           | Description    | Type           |
+| -------------- | -------------- | -------------- |
+| --kafka-bootstrap-servers= | List of Kafka brokers as a comma separated list of broker `host` or `host:port`. | `[string]` |
+| --stream-transaction-conflict-retries=30 | Number of times to retry a conflicting transaction of a stream | `[uint32]` |
+| --stream-transaction-retry-interval=500 | The interval to wait (measured in milliseconds) before retrying to commit a conflicting transaction | `[uint32]` |
```
[the remark this answers is not in the document]
Comment (kostasrim) [reply docs/reference-guide/configuration.md:60]: nice

**Review by antonio2368 — APPROVED** [review]

**Review by antaljanosbenjamin — APPROVED** [review]
